internal/commands/app/describe_test.go

```diff
@@ -0,0 +1,380 @@
+				},
+				GraphQL: realm.GraphQLSummary{
+					URL: "https://api-base.url/api/client/v2.0/app/todo-abcde/graphql",
+					CustomResolvers: []string{
+						"type.fieldName",
+					},
+				},
+				Environment: "development",
+				EventSubscription: []realm.EventSubscriptionSummary{
+					{
+						Name:    "trigger1",
+						Type:    "DATABASE",
+						Enabled: true,
+					},
+				},
+			}, nil
+		}
+
+		cmd := &CommandDescribe{inputs: describeInputs{cli.ProjectInputs{App: "test-app-abcd"}}}
+		assert.Nil(t, cmd.Handler(nil, ui, cli.Clients{Realm: realmClient}))
+
+		assert.Equal(t, `01:23:45 UTC INFO  App Description
+{
+  "client_app_id": "todo-abcde",
+  "name": "todo",
+  "realm_url": "https://admin-base.url/groups/123/apps/456/dashboard",
+  "data_sources": [
+    {
+      "name": "mongodb-atlas",
+      "type": "mongodb-atlas",
+      "data_source": "Cluster0"
+    },
+    {
+      "name": "mongodb-datalake",
+      "type": "datalake",
+      "data_source": "DataLake0"
+    },
+    {
+      "name": "mdb1",
+      "type": "mongodb",
+      "data_source": ""
+    }
+  ],
+  "http_endpoints": [
+    {
+      "name": "http"
+    }
+  ],
+  "services": [
+    {
+      "name": "tw1",
+      "type": "twilio"
+    }
+  ],
+  "auth_providers": [
+    {
+      "name": "oauth2/google",
+      "type": "oauth2/google",
+      "enabled": true
+    },
+    {
+      "name": "oauth2/facebook",
+      "type": "oauth2/facebook",
+      "enabled": true
+    }
+  ],
+  "custom_user_data": {
+    "enabled": true,
+    "data_source": "Cluster0",
+    "database": "db1",
+    "collection": "col1",
+    "user_id_field": "id"
+  },
+  "values": [
+    "value1",
+    "value2"
+  ],
+  "hosting": {
+    "enabled": true,
+    "status": "setup_ok",
+    "url": "https://hosting.domain/"
+  },
+  "functions": [
+    {
+      "name": "func1",
+      "path": "func1.js"
+    },
+    {
+      "name": "func2",
+      "path": "nested/func2.js"
+    }
+  ],
+  "sync": {
+    "state": "enabled",
+    "data_source": "Cluster0",
+    "database": "db1",
+    "development_mode_enabled": true
+  },
+  "graphql": {
+    "url": "https://api-base.url/api/client/v2.0/app/todo-abcde/graphql",
+    "custom_resolvers": [
+      "type.fieldName"
+    ]
+  },
+  "environment": "development",
+  "event_subscription": [
+    {
+      "name": "trigger1",
+      "type": "DATABASE",
+      "enabled": true
+    }
+  ]
+}
+`, out.String())
+	})
+
+	t.Run("should error if apps not found", func(t *testing.T) {
+		realmClient := mock.RealmClient{}
+
+		realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+			return []realm.App{}, nil
+		}
+
+		cmd := &CommandDescribe{inputs: describeInputs{cli.ProjectInputs{App: "test-app"}}}
+		assert.Equal(t, cli.ErrAppNotFound{App: "test-app"}, cmd.Handler(nil, nil, cli.Clients{Realm: realmClient}))
+	})
+
+	for _, tc := range []struct {
+		description   string
+		inputs        describeInputs
+		groups        []atlas.Group
```

Comment:
i think we should probably remove this test table, and just write the individual error test cases separately.  some of these test case struct fields appear to be unused, and the test cases could be elaborated a bit more to explain what each test is actually doing (i.e. `"should return an error when finding atlas groups fails"`, etc)